internal/dashboard/business/ac/enforcers.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright (c) 2024 OceanBase
+ob-operator is licensed under Mulan PSL v2.
+You can use this software according to the terms and conditions of the Mulan PSL v2.
+You may obtain a copy of Mulan PSL v2 at:
+         http://license.coscl.org.cn/MulanPSL2
+THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND,
+EITHER EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT,
+MERCHANTABILITY OR FIT FOR A PARTICULAR PURPOSE.
+See the Mulan PSL v2 for more details.
+*/
+
+package ac
+
+import (
+	"strings"
+
+	"github.com/gin-contrib/sessions"
+	"github.com/gin-gonic/gin"
+)
+
+type EnforceFunc func(c *gin.Context) (bool, error)
+
+// groups of domain, resource and action
+// if resource is prefixed with ":", it means a parameter in the path
+// if resource contains "+", it means a list of resources, like :namespace+:name
+func PathGuard(perms ...string) EnforceFunc {
```

Comment:
If you need to receive more than one resource permissions, pass them in a structured way

Reply:
maybe there's no need to support mulltiple resource permissions

Reply:
It's painful to put structs in API router registration. :( By now there is only one pair of domain, object and action used in practice.